NMS-14621: Quick Start - Visualize Data (#5353)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/admin/webui/surveillance-view.adoc

[[surveillance-view]]
= Surveillance View

> When monitoring a large network that contains devices of different priorities, you may want to see how the "whole system" is working at a glance.

Reviewer: ```suggestion
When monitoring a large network that contains devices of different priorities, you may want to see how the whole system is working at a glance.
```

> The Surveillance View feature aims to do that.

Reviewer: ```suggestion
Use the Surveillance View feature to do so.
```

| Vanity             | 4 of 10 | 0 of 10
|===

In this way, the Surveillance View can provide a quick idea of where trouble might be occurring.
The matrix display allows for a significantly higher aggregation of data than a simple list.
The Surveillance View also shows nodes rather than services--an important thing to keep in mind when you look at categories.

When monitoring your network, you want to know how many servers or services are experiencing issues.

.Surveillance View displaying three categories; one category is experiencing an outage
image::surveillance-view/01_surveillance-view.png[]

> Visual status indicators are defined as follows:

Reviewer: ```suggestion
The following defines the visual status indicators:
```

> Visual status indications are defined as follows:

Reviewer: ```suggestion
Visual status indicators are defined as follows:
```

* *Green:* no services down.
* *Yellow:* one service down.
* *Red:* more than one service down.

> The Surveillance View also serves as the foundation of the <<deep-dive/admin/webui/dashboard.adoc, Dashboard>>.

Author: I'll test it out locally and update it. :)

== Default configuration

Surveillance Views are defined in `$\{OPENNMS_HOME}/etc/surveillance-views.xml`.

NOTE: `surveillance-views.xml` can be modified using a text editor.

> It is re-evaluated every time the Surveillance View page is loaded.

Reviewer: What is re-evaluated every time the SV is loaded? Should it be:

```suggestion
{page-component-title} re-evaluates the file every time the Surveillance View page is loaded.
```

> As such, changes to the file do not require {page-component-title} to be restarted.

Reviewer: ```suggestion
Changes to the file do not require you to restart {page-component-title}.
```

  </views>
</surveillance-view-configuration>
----

WARNING: The old `report-category` attribute is deprecated and is no longer supported.

== Custom configuration

> The Surveillance View can be modified using the Surveillance View Configurations editor:

Reviewer: ```suggestion
Use the Surveillance View Configurations editor to modify the Surveillance View:
```

. Click the *Gear* symbol at the top-right of the screen.
. Under Additional Tools, click *Surveillance Views Configuration*.

The page displays an overview of configured Surveillance Views and lets you edit, remove, and preview them.
You can also select a default Surveillance View by selecting the Default box beside its name.

> .Surveillance View Configuration UI, with three entries; the `default` configuration is designated as the default View

Reviewer: ```suggestion
.Surveillance View Configuration UI, with three entries; the `default` configuration is designated as the default view
```

image::surveillance-view/02_surveillance-view-config-ui.png[]

When you configure a Surveillance View, you must define its title and the time (in seconds) between refreshes.
You can also add and delete columns and rows, and reorder them as appropriate by clicking the *Up Arrow* and *Down Arrow* symbols.

.Surveillance View Configuration dialog
image::surveillance-view/03_surveillance-view-config-ui-edit.png[]

Editing row or column definitions requires you to choose a unique label and at least one {page-component-title} category.

> When you are finished, you can click *Save* to persist the configuration settings.

Reviewer: ```suggestion
When you are finished, click *Save* to persist the configuration settings.
```

You can also click *Cancel* to close the dialog without saving.

== Categorizing nodes

Follow the steps below to categorize a node in the Surveillance View:

. Click menu:Info[Nodes] in the top menu bar.
. Click *Edit* beside the Surveillance Category Memberships section title.
. Select the check box at the bottom of the screen to enable category membership controls.
+

> WARNING: Any changes made to a node that was provisioned through a requisition will be rolled back the next time the node is synchronized or manually rescanned.

Reviewer: It would be good if this didn't happen, but that's a development thing.

. Edit the category memberships:
** Select categories from the Available Categories list and click *Add >>* to add them to the node.
** Select categories from the Categories on Node list and click *Remove <<* to remove them from the node.

Your changes are automatically saved.

== Creating views for users and groups

> You can create user and group names for Surveillance Views.

Reviewer: It's not actually clear to me how to do this. Can we add this information?

> When the Surveillance View page is loaded, the first of the following criteria to be fulfilled determines the view that is displayed:

Reviewer: ```suggestion
When the Surveillance View page loads, the first of the following criteria to be fulfilled determines the view that is displayed:
```

Editing row or column definitions requires you to choose a unique label and at least one {page-component-title} category.

When you are finished, click *Save* to persist the configuration settings.
You can also click *Cancel* to close the dialog without saving.

== Categorizing nodes

Follow the steps below to categorize a node in the Surveillance View:

. Click menu:Info[Nodes] in the top menu bar.
. Click *Edit* beside the Surveillance Category Memberships section title.
. Select the check box at the bottom of the screen to enable category membership controls.
+
WARNING: Any changes made to a node that was provisioned through a requisition will be rolled back the next time the node is synchronized or manually rescanned.

. Edit the category memberships:
** Select categories from the Available Categories list and click *Add >>* to add them to the node.
** Select categories from the Categories on Node list and click *Remove <<* to remove them from the node.

Your changes are automatically saved.

== Creating views for users and groups

You can create Surveillance Views for only specific users, or only users in specific groups.

> For example, to create a view that can only be accessed by the user `drv4doe`, edit `$\{OPENNMS_HOME}/etc/surveillance-view.xml` and add a definition for the user:

Reviewer: ```suggestion
For example, to create a view that only the user `drv4doe` can access, edit `$\{OPENNMS_HOME}/etc/surveillance-view.xml` and add a definition for the user:
```

[source, xml]
----
<view name="drv4doe" refresh-seconds="300" >
  <rows>
    <row-def label="Servers" >
      <category name="Servers"/>
    </row-def>
  </rows>
  <columns>
    <column-def label="PROD" >
      <category name="Production" />
    </column-def>
    <column-def label="TEST" >
      <category name="Test" />
    </column-def>
  </columns>
</view>
----

> The same can be done for group names.

Reviewer: ```suggestion
You can do the same for group names.
```

> NOTE: The definition should be added within the `<views>` tag, above the default view.

Reviewer: ```suggestion
NOTE: You should add the definition within the `<views>` tag, above the default view.
```

> When the Surveillance View page loads, the first of the following criteria to be fulfilled determines the view that is displayed:

Reviewer: ```suggestion
When the Surveillance View page loads, the first of the following criteria to be fulfilled determines the view it displays:
```
